### PR TITLE
AK + Kernel: Utilize KResult + AK::Userspace<T> in the ioctl interface

### PR DESCRIPTION
**Pull request by bgianfo**
In the interest of keeping the Kernel IOCTL API updated with the stye of the rest of the system, switch to returning `KResult` and taking `Userspace<void*>`. 

- **Kernel: Modify the IOCTL API to return KResult**

    The kernel has been gradually moving towards KResult from just bare
    int's, this change migrates the IOCTL paths.

-  **Kernel+LibC: Use argument for TIOCGPGRP ioctl value**

    In preparation for modifying the Kernel IOCTL API to return KResult
    instead of int, we need to fix this ioctl to an argument to receive
    it's return value, instead of using the actual function return value.

- **Kernel: Utilize AK::Userspace<T> in the ioctl interface**

    It's easy to forget the responsibility of validating and safely copying
    kernel parameters in code that is far away from syscalls. ioctl's are
    one such example, and bugs there are just as dangerous as at the root
    syscall level.

    To avoid this case, utilize the AK::Userspace<T> template in the ioctl
    kernel interface so that implementors have no choice but to properly
    validate and copy ioctl pointer arguments.

- **AK: Add copy_to span method for AK::MACAddress**

### CODE REVIEW

File: AK/MACAddress.h [hunk 1]
```diff
@@ -84,6 +84,11 @@ class [[gnu::packed]] MACAddress {
         return all_of(m_data.begin(), m_data.end(), [](const auto octet) { return octet == 0; });
     }
 
+    void copy_to(Span<u8> destination)
```
Comment (alimpfard) [inline AK/MACAddress.h:87]: `s/Span<u8>/Bytes`?

Comment (bgianfo) [reply AK/MACAddress.h:87]: Sure

Comment (bgianfo) [reply AK/MACAddress.h]: fixed

File: Kernel/Graphics/VirtIOGPU/FrameBufferDevice.cpp [hunk 2]
```diff
@@ -140,62 +140,63 @@ void FrameBufferDevice::set_buffer(int buffer_index)
...
         resolution.height = height();
         if (!copy_to_user(user_resolution, &resolution))
-            return -EFAULT;
-        return 0;
+            return EFAULT;
+        return KSuccess;
     }
     case FB_IOCTL_SET_BUFFER: {
-        auto buffer_index = (int)arg;
+        auto buffer_index = (int)arg.unsafe_userspace_ptr();
```
Comment (alimpfard) [inline Kernel/Graphics/VirtIOGPU/FrameBufferDevice.cpp:177]: This needs a cast to `FlatPtr` before a cast to `int`.

Comment (bgianfo) [reply Kernel/Graphics/VirtIOGPU/FrameBufferDevice.cpp:177]: Why's that?

**Review by alimpfard — APPROVED** [review]

**Pull request merged**